src/providers/spatialite/qgsspatialiteconnection.cpp

```diff
@@ -563,38 +563,34 @@ bool QgsSpatiaLiteConnection::isRasterlite1Datasource( sqlite3 *handle, const ch
   int rows;
   int columns;
   bool exists = false;
-  char table_raster[4192];
-  char sql[4258];
 
-  strncpy( table_raster, table, sizeof table_raster );
-  table_raster[sizeof table_raster - 1] = '\0';
-
-  const size_t len = strlen( table_raster );
-  if ( strlen( table_raster ) < 9 )
+  QString tableRaster = QString::fromUtf8( table );
+  if ( tableRaster.size() < 9 )
```

Comment:
this check can be removed as it is implicit by the next one checking for endsWith("_metadata")